### PR TITLE
LITTIL-164 - Protect private info

### PR DESCRIPTION
**Pull request by roelandkulk**

### CODE REVIEW

File: src/main/java/org/littil/api/guestTeacher/service/GuestTeacherService.java [hunk 1]
```diff
@@ -35,18 +37,25 @@ public class GuestTeacherService {
     private final UserService userService;
     private final UserMapper userMapper;
     private final AuthenticationService authenticationService;
+    private final TokenHelper tokenHelper;
 
 
     public List<GuestTeacher> getTeacherByName(@NonNull final String name) {
-        return repository.findBySurnameLike(name).stream().map(mapper::toDomain).toList();
+        UUID userId = tokenHelper.getCurrentUserId();
```
Comment (pschildkamp) [inline src/main/java/org/littil/api/guestTeacher/service/GuestTeacherService.java:44]: This method now only returns a teacher object which correnspons with the current user. From my understanding, we had to filter the properties of the teacher, not the teacher itself. For example:

- When fetching teacher details which are attached to my user I will get all the properties available
- When fetching a teacher from another user on the platform I only get properties that are public available. We want to mask address properties, for example, they are only needed for coordinates.

File: src/main/java/org/littil/api/guestTeacher/service/GuestTeacherService.java [hunk 2]
```diff
@@ -85,9 +94,13 @@ public void deleteTeacher(@NonNull final UUID id, UUID userId) {
     }
 
     private GuestTeacher update(@Valid GuestTeacher guestTeacher) {
+        UUID userId = tokenHelper.getCurrentUserId();
         GuestTeacherEntity entity = repository.findByIdOptional(guestTeacher.getId())
                 .orElseThrow(() -> new NotFoundException("No Teacher found for Id"));
 
+        if (entity.getUser() != null && !entity.getUser().getId().equals(userId)) {
```
Comment (pschildkamp) [inline src/main/java/org/littil/api/guestTeacher/service/GuestTeacherService.java:101]: You already know the user you want to update and your current userId in the resource. Would this be a better place block this request?

**Review by pschildkamp — DISMISSED** [review]

File: src/main/java/org/littil/api/guestTeacher/service/GuestTeacherService.java (hunk 2 above)
Comment (roelandkulk) [reply src/main/java/org/littil/api/guestTeacher/service/GuestTeacherService.java:102]: In the resource we receive a GuestTeacherPostResource which doesn't include the user. Only in the service we fetch the GuestTeacherEntity which includes the user.

File: src/main/java/org/littil/api/guestTeacher/service/GuestTeacherService.java (hunk 1 above)
Comment (roelandkulk) [reply src/main/java/org/littil/api/guestTeacher/service/GuestTeacherService.java]: I have made some changes to apply this feedback.

**Review by MarcelWildenburg — COMMENTED** [review]
No comments on changes in main code. Final review can be done after update of unit test.

**Review by MarcelWildenburg — APPROVED** [review]

**Pull request merged**